src/dev/src/app/tree-view/utils/async-infinite-tree.ts

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright (c) 2016-2018 VMware, Inc. All Rights Reserved.
+ * This software is released under MIT license.
+ * The full license information can be found in LICENSE in the root directory of this project.
+ */
+
+import { ClrSelectedState } from '@clr/angular';
+
+import { InfiniteTree } from './infinite-tree';
+import { Observable, timer } from 'rxjs';
+import { map } from 'rxjs/operators';
+
+export class AsyncInfiniteTree {
+  constructor(width: number, latency = 0) {
```

Comment:
nit: Can we put a 100ms delay or something very small here just so it feels a little bit async?

Reply:
Sure, I'll make the default delay 100ms. 👍 

Note that the demos actually set the latency to 500ms, so the spinners are clearly visible.